### PR TITLE
Fix incorrect Java equals comparison of Uuid by reference

### PR DESCRIPTION
**Pull request by alok123t**
Previously, we were comparing uuid by reference, which could be incorrect. This PR replaces to do the comparison by value instead

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Review by jolshan — APPROVED** [review]
Thanks for fixing the log too :)

**Review by chia7712 — COMMENTED** [review]
@alok123t nice finding. Could you please add a test for it?

**Comment by chia7712** [comment]
the following links have similar issue. Should we fix them in this PR or I file a follow-up to fix them?

https://github.com/apache/kafka/blob/61baa7ac6bb871797197d9289a848a0d4f587ced/clients/src/main/java/org/apache/kafka/clients/admin/KafkaAdminClient.java#L1704

https://github.com/apache/kafka/blob/e02ffd852fae7c1d2681621be7eb888e3805e027/tools/src/main/java/org/apache/kafka/tools/TopicCommand.java#L299

**Comment by alok123t** [comment]
@jolshan @chia7712 randomly looking at the code base, i think this is prevalent in multiple places. Shall I update them all in this PR and also create a ticket?

**Comment by alok123t** [comment]
i did smth similar to https://stackoverflow.com/a/73535824 to find all usages in the codebase

File: core/src/main/scala/kafka/server/AlterPartitionManager.scala [hunk 1]
```diff
@@ -270,7 +270,7 @@ class DefaultAlterPartitionManager(
 
     inflightAlterPartitionItems.groupBy(_.topicIdPartition.topic).foreach { case (topicName, items) =>
       val topicId = items.head.topicIdPartition.topicId
-      canUseTopicIds &= topicId != Uuid.ZERO_UUID
+      canUseTopicIds &= !topicId.equals(Uuid.ZERO_UUID)
```
Comment (chia7712) [inline core/src/main/scala/kafka/server/AlterPartitionManager.scala]: IIRC, it calls the equals method on the first object in Scala.

**Review by chia7712 — COMMENTED** [review]
@alok123t thanks for adding more fixes. one small question left. PTAL

File: core/src/main/scala/kafka/server/AlterPartitionManager.scala (hunk 1 above)
Comment (alok123t) [reply core/src/main/scala/kafka/server/AlterPartitionManager.scala]: right, we don't need to do this for Scala files, reverted change in https://github.com/apache/kafka/pull/15707/commits/e206a2cfdfa955b3ab004b5644e28e6fa23999b9

**Review by chia7712 — APPROVED** [review]
LGTM

**Pull request merged**